### ISSUE TITLE
My private repositories not seen when clicking "Create App"

### ISSUE THREAD

**Issue by minagawah** [eclipsesource/tabris-js#917]
Hi. I was looking at [the plan description page](https://tabrisjs.com/pricing/) for Tabris.js,
and it says that I can specify one of my private GitHub repositories for cloud builds.
I thought it's a pretty good deal, so I gave a try.
From my dashboard, as I open *"My Apps"* page, I can see *"Create App"* button there.
But, when I click the button, I see a list of my public repos alright, but not the private ones...
I tried the synch button several times, but no luck...
I'm sure I am on the *Developer* plan, which is required for the private repo build.
Is there anything I need to do for my GitHub account? Or, are there any optional settings on dashboard?

**Comment by gosia-eclipsesource** [comment]
Hi @minagawah, sorry about the issue. Could you please check in your [application GitHub settings](https://github.com/settings/applications) whether Tabris.js has the following permissions?
-  Read org and team membership
-  Full control of private repositories
-  Access user email addresses (read-only)

**Comment by sand123** [comment]
Last time I had same issue, I revoked Tabris.js access on https://github.com/settings/applications and relogin on tabris.js - that solved private repo issues

**Comment by gosia-eclipsesource** [comment]
Thanks @sand123. @minagawah, could you try if the above works for you?

**Comment by minagawah** [comment]
Hi @sand123 and @gosia-eclipsesource
Thanks. That makes a perfect sense!
When I granted access to Tabris.js app, I wasn't a paid GitHub user yet.
Checking [oAuth settings page](https://github.com/settings/connections/applications) on my GitHub for Tabris.js app,
I noticed **"Full control of private repositories"** was missing.

OK. Here is what I just did:

1. I revoked Tabris.js app on my GitHub.
1. I removed the cache and data on my android device for Tabris.js app.
1. Tabris.js app is now asking me to sign-in with my GitHub account, so I did.
1. Then it asks me if I allow the access.
1. Feeling weird for it still doesn't ask me for access to my private repo, I clicked Yes anyway.
1. Was expecting "Create App" button showing my private repos this time, but it didn't....
1. Checking again [oAuth settings page](https://github.com/settings/connections/applications), it still does not have the permission concerned...

GitHub may have some other settings page, so that I can allow the third-party apps for the permission.
I will be struggling few more hours on this.
Thanks @sand123 and @gosia-eclipsesource anyways!
I will let you know for updates if I ever find ones...

**Comment by minagawah** [comment]
I'm not sure if this is relevant to the issue but I found an interesting comment:

http://stackoverflow.com/questions/21907278/github-api-using-repo-scope-but-still-cant-see-private-repos#21909135

For the program hosted on "My App" page on Tabris.js dashboard, it is executing an ajax call to the local API, namely, `/api/v1/me/repositories` which simply returns all the repositories belong to the user.
@gosia-eclipsesource , is it possible to ask the server guy if this one applies?
If the program is using `/users/someusername/repos` instead of `/user/repos` ?

In a meanwhile, I will still keep on searching the way out...

**Comment by minagawah** [comment]
Nah.. if there's a problem, the server has nothing to do with it, but should be with either on the client side or on the settings I have on GitHub...
Never mind about the above comment.

**Comment by minagawah** [comment]
A bit of update. I forgot to re-login from the Tabris.js dashboard.
So, I logged out, and did the re-login.
I see the website asks me two of the above mentioned permissions except for the private repo control...
I now realized both client and server are accessing the GitHub API.
Anyways, still looking...

**Comment by gosia-eclipsesource** [comment]
We have looked into it, and this is an issue with our GitHub scopes implementation. We are working on a solution.

There is an interim solution, so for other users who experience this, please let us know (care@tabrisjs.com) and we will fix it on a per-case basis.

**Comment by minagawah** [comment]
@gosia-eclipsesource 
Thank you. For now, I will make my project public.

**Comment by jandrieu** [comment]
I am having this problem. I emailed care@tabrisjs.com and haven't heard back. how can I get access to my private repo?

**Comment by minagawah** [comment]
@jandrieu , I assume they are working pretty hard fixing the problem. Since I paid GitHub only for this feature Tabris.js provides, I know how painful it is for you as well.... Let us hope they will raise the priority on this issue for there must be other developers like us.

**Comment by gosia-eclipsesource** [comment]
Yes, we understand it's a big issue and we will do what we can to fix it soon.
You should both have gotten an e-mail from us. @minagawah please check my e-mail from 8 July if you haven't seen it yet. Thanks

**Comment by jandrieu** [comment]
They got a workaround for me, but then they deleted the issue
#917.  Weird.
 
 
On Tue, Jul 12, 2016, at 05:26 PM, Hiroki Minagawa wrote:
> @jandrieu[1] , I assume they are working pretty hard fixing the
> problem. Since I paid GitHub only for this feature Tabris.js provides,
> I know how painful it is for you as well.... Let us hope they will
> raise the priority on this issue for there must be other developers
> like us.
> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].
>
 
--
Joe Andrieu
joe@andrieu.net
+1(805)705-8651
http://blog.joeandrieu.com
 

Links:

  1. https://github.com/jandrieu
  2. https://github.com/eclipsesource/tabris-js/issues/917#issuecomment-232221245
  3. https://github.com/notifications/unsubscribe/AALZzxRsnVLqfe_L3FqR6Pq1G8Ce-Lqeks5qVDDHgaJpZM4JG-Z5

**Comment by jonaspm** [comment]
I have the same problem

**Comment by gosia-eclipsesource** [comment]
Hi @jonaspm, sorry, we are still working on it. We sent you a note yesterday, in case you haven't seen it please check your email.

**Comment by jonaspm** [comment]
Yeah, thank you so much! I can now see my private repositories.

I hope you can fix it soon so no one gets scared by this.


Thanks also for creating this amazing technology. Very useful!

I am currently working on an app for my university and working with TabrisJS has been very straightforward [😊]


Cheers,

Jonás Perusquía Morales


________________________________

**Comment by camstuart** [comment]
So has a solution been found? same issue here

**Comment by scmorse** [comment]
I was having this exact issue. I got around it by adding a [read-only deploy key](https://github.com/blog/2024-read-only-deploy-keys), and then using the "Create App" by "Git URL" option:

<img width="917" alt="screen shot 2017-01-10 at 12 10 43 am" src="https://cloud.githubusercontent.com/assets/4519249/21794559/611149b6-d6c9-11e6-8e3d-8848979bf67d.png">

And [here's the link](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/) for instructions on creating a new SSH key. You add the .pub file contents into GitHub settings so that GitHub can recognize the deploy key, and then add the private part into the "SSH Private Key" text box.

**Comment by gosia-eclipsesource** [comment]
Just a note that Git URL is currently only available on the Pro plan. The private repo implementation is only broken on the free Developer plan (we are still working on this issue).

**Comment by Zeldaze** [comment]
Any update to this? I too am missing private repository access on my Free account for testing purposes.
